### PR TITLE
Linker - Added import linking

### PR DESCRIPTION
**Pull request by danielstro**

### CODE REVIEW

File: packages/schema-extract/src/extract-schema.ts [hunk 1]
```diff
@@ -4,18 +4,31 @@ import typescript from 'typescript';
 import {LocalFileSystem} from 'kissfs';
 import {transform} from './file-transformer';
 import {createHost} from './isomorphc-typescript-host';
+import {SchemaLinker} from './file-linker';
 
-export function* extractSchema(basePath: string, filesGlob: string) {
+export function* extractSchema(basePath: string, filesGlob: string, linked = false) {
```
Comment (nadavwix) [inline packages/schema-extract/src/extract-schema.ts]: the linked schema should be moved to a different function.

it requires the name of the export to link

File: packages/schema-extract/src/file-linker.ts [hunk 2]
```diff
@@ -1,140 +1,193 @@
...
+        this.sourceFile = sourceFile;
+        const schema = transform(this.checker, sourceFile, file, this.projectPath);
+        let entity;
+        if (schema.definitions) {
+            entity = schema.definitions[entityName];
+        } else if (schema.properties) {
+            entity = schema.properties[entityName];
+        }
         if (!entity) {
             return {};
```
Comment (nadavwix) [inline packages/schema-extract/src/file-linker.ts]: should return common/unknown

File: packages/schema-extract/test/linker/imports.spec.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                something: {
+                    type: 'string'
+                }
+            },
+            required: ['something']
+        };
+        expect(res).to.eql(expected);
+    });
+
+    it('should link imported type from an outside package', async () => {
```
Comment (nadavwix) [inline packages/schema-extract/test/linker/imports.spec.ts]: should add a test that shows we work on a linked d.ts file

File: packages/schema-extract/src/extract-schema.ts (hunk 1 above)
Comment (danielstro) [reply packages/schema-extract/src/extract-schema.ts]: I created a new function and changed it back. Anyway the cli command doesn't work properly yet and I will fix it separately

File: packages/schema-extract/src/file-linker.ts (hunk 2 above)
Comment (danielstro) [reply packages/schema-extract/src/file-linker.ts]: Fixed

**Pull request merged**